### ISSUE TITLE
[Suggestion] Chunk Materials

### ISSUE THREAD

**Issue by Ayliroe** [BLaZeKiLL/UE5VoxelTutorial#14]
Here's an implementation I found working you can add to your tutorial:

- ChunkBase.h:
![image](https://user-images.githubusercontent.com/106384472/175832039-30fb48a7-510e-4ddc-ae92-2693c89c73f6.png)
(A blueprint-accessible array for materials, plus arrays for the material struct and vertices instead of single values)

- ChunkBase.cpp:
![image](https://user-images.githubusercontent.com/106384472/175832090-c8665586-8f50-4cb9-9ff7-12764034ebcb.png)
(On clearing, empty the arrays then refresh using the total amount of materials, not blocks)

- NaiveChunk.cpp, in GenerateMesh():
![image](https://user-images.githubusercontent.com/106384472/175832156-13f40e02-b419-4dbf-95a7-9e743b00c7d1.png)
(Get the material for the current block's face, minus 2 to skip null and air enums, clamp as a safety, then pass it to CreateFace() )

- NaiveChunk.cpp, in CreateFace():
![image](https://user-images.githubusercontent.com/106384472/175832228-6ba00f86-86e9-4bba-8f68-6684717e0efb.png)
(Switch all mesh structs and the vertex int to the new arrays)

- GreedyChunk.cpp, GenerateMesh():
![image](https://user-images.githubusercontent.com/106384472/175832276-09533364-4b04-4b3a-b178-b9cc4bbd7285.png)
(Same as NaiveChunk, but using CurrentMask.Block for the face's material)

- ApplyMesh():
![image](https://user-images.githubusercontent.com/106384472/175832346-ec410969-f428-46d2-a438-c749dc39e9f3.png)
(First clear all previous sections, then generate a new one for each material on the chunk)

Basically the same code as before, but generates one unique section for each different block material. Straightforward to scale-up (just add materials to the array) and supports both naive/greedy meshing. Maybe you had something in mind too of the like.

**Comment by jorgeEsat** [comment]
What is GetID on the third image?

**Comment by AnasTheDev** [comment]
@Ayliroe Hey man. I sadly cannot seem to get your code to work. mind adding me on discord so I can bother you there haha?
No seriously please add me on discord I really need the help here is my username "anasdev"

**Comment by PeterKadritzke** [comment]
Code is bad documented and causes out of bounds array. Git is literally a version controll Software that allows you to make branches of diffrent types of versions.
@AnasTheDev Did he help you, or did you get it otherwise?
